### ISSUE TITLE
Using Sprite3D instead of sprite?

### ISSUE THREAD

**Issue by elvisish** [smix8/Godot3DTargetTracker#1]
Love the example, I can actually understand how this works compared to other offscreen trackers I've seen! Is it possible to use this with a Sprite3D? I can't quite figure out how to get the camera position x/y for the edges in 3D since this uses unproject_position.

**Comment by smix8** [comment]
You could do the same but instead of placing the Sprite at the final position project this 2D screenposition back to the 3D world with e.g. Camera.project_position() and move your spatial from the Sprite3D at this 3D position.

**Comment by elvisish** [comment]
> You could do the same but instead of placing the Sprite at the final position project this 2D screenposition back to the 3D world with e.g. Camera.project_position() and move your spatial from the Sprite3D at this 3D position.

Thanks for the suggestion! Trying something like this, it's not working yet for some reason but is this the concept?
```swift
	var viewport_rect = get_viewport().get_visible_rect()
	var target_2d_position: Vector2 = cam.unproject_position(transform.origin)
	
	pos2d.position.x = clamp(target_2d_position.x, screen_border_offset.x, viewport_rect.size.x - screen_border_offset.x)
	pos2d.position.y = clamp(target_2d_position.y, screen_border_offset.y, viewport_rect.size.y - screen_border_offset.y)
	
	transform.origin.x = cam.project_position(pos2d.position,1.0).x
	transform.origin.z = cam.project_position(pos2d.position,1.0).z
	
	if viewport_rect.has_point(target_2d_position):
		target_2d_position = cam.unproject_position(start_pos)
```

**Comment by smix8** [comment]
`sprite_3d.global_transform.origin = cam.project_position(pos2d.position,cam.distance_to(global_transform.origin))`

The 2nd parameter for project_position is the project distance from the camera into the 3D scene. With just 1.0 unit the mesh will be placed so close to the camera and might get already culled. If you use the value from camera.distance_to(sprite_3d) it might not be perfect but at least should work that you can finetune it. Depending on your node tree you might need to use global_transform instead of transform as transform alone is relative to the parent node.

**Comment by elvisish** [comment]
> sprite_3d.global_transform.origin = cam.project_position(pos2d.position,cam.distance_to(global_transform.origin))

Thanks! Since I'm tracking a point that doesn't change, I set the distance in ready:
`onready var cam_distance = cam.transform.origin.distance_to(start_pos)`
then the distance is the same, since it's top-down and I only need the z-axis distance. Then I did as you suggested and it works great!
`sprite3d.global_transform.origin = cam.project_position(pos2d.position,cam_distance)`

Thanks for the global_transform.origin vs. transform.origin tip, I didn't realize they were different!
